HIP/hip-1195.md

```diff
@@ -0,0 +1,834 @@
+`SLOAD` and `SSTORE` operations with its storage, and so on.
+
+If the `0.0.A.1` hook is a pure EVM hook, it executes in an extremely restricted mode. Not only is the initial frame
+marked **static**, prohibiting all state-changing operations; but the network also disables the `PREVRANDAO`, `SLOAD`,
+and `CALL` opcodes. That is, a pure hook cannot do anything but apply a pure function to its input data.
+
+Although users will doubtless uncover a very wide range of use cases for EVM hooks, we expect the single most common
+type of hook contract to implement a single external method that reverts when not executed by the network _as a hook_.
+That is,
+
+```solidity
+contract HookContract {
+    function hookFunction(address installer) external payable {
+        // Revert if we are not executing a DELEGATECALL from the hook system contract
+        require(address(this) == 0x16d, "Contract can only be called as a hook");
+        // Continue executing as a hook on behalf of the installer address
+    }
+}
+```
+
+Because a hook is conceptually a function of the installing account, when it calls a Hedera system contract, the system 
+contract will see that account's address as the sender. If a hook calls another contract, it is valid for that contract 
+to call back to the hook at address `0x16d`.
+
+### Core HAPI protobufs
+
+A hook's extension point is one of an enumeration which now includes only the account allowance hook,
+```protobuf
+/***
+ * The Hiero extension points that accept a hook.
+ */
+enum HookExtensionPoint {
+  /**
+   * Used to customize an account's allowances during a CryptoTransfer transaction.
+   */
+  ACCOUNT_ALLOWANCE_HOOK = 0;
+}
+```
+
+Users install hooks by setting a `HookInstall` field on an appropriate create or update transaction. This message is,
+
+```protobuf
+/***
+ * How to install a hook.
+ */
+message HookInstall {
+  /**
+   * The extension point for the hook.
+   */
+  HookExtensionPoint extension_point = 1;
+
+  /**
+   * The entity index to install the hook at.
+   */
+  uint64 index = 2;
+
+  /**
+   * The hook implementation.
+   */
+  oneof hook {
+    /**
+     * A hook programmed in EVM bytecode that does not require access to state
+     * or interactions with external contracts.
+     */
+    PureEvmHook pure_evm_hook = 3;
+    /**
+     * A hook programmed in EVM bytecode that may access state or interact with
+     * external contracts.
+     */
+    LambdaEvmHook lambda_evm_hook = 4;
+  }
+
+  /**
+   * If set, a key that that can be used to remove or replace the hook; or (if
+   * applicable, as with a lambda EVM hook) perform transactions that customize
+   * the hook.
+   */
+  proto.Key admin_key = 5;
+
+  /**
+   * The charging pattern for the hook.
+   */
+  oneof charging_spec {
+    /**
+     * The payer of the transaction calling the hook pays for its execution.
+     */
+    CallerPays caller_pays = 6;
```

Comment:
The HIP does not define the `CallerPays` type. In addition, instead of defining a new type, I suggest to use `bool` for `caller_pays`.

Reply:
Removed per Leemon's comments in TSC.